jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -254,6 +255,27 @@ public String getVersion() {
     return project.getVersion().toString();
   }
 
+  void validateBaseImageVersion(@Nullable String baseImage) {
+    if (!PluginConfigurationProcessor.usingDefaultBaseImage(baseImage)) {
+      return;
+    }
+    Map<String, ?> javaCompile = project.getProperties();
```

Comment:
Few comments about this section
1. It's possible that we only need to worry about `targetCompatibility` as that is the byte code level of the generated code?
2. The default value when neither is set is = JavaVersion.current() I think.
3. We can probably query the `JavaPluginConvention` to get these values, instead of parsing project properties.
4. I think `kotlin` might have special behavior for this, but I'm not sure.
5. I think you can use `JavaVersion` to do all your math.
    - For parsing: `JavaVersion.toVersion()` or `JavaVersion.valueOf()`
    - For compating: `parsedJavaVersion.isJava8Compatible()`

Reply:
Thanks, is there an easy Maven equivalent? `JavaVersion` is gradle-only.

Reply:
Yeah I don't know, I remember we had to do some shenanigans to get the version from maven. It might be even more complicated now. Let me look around.

Reply:
So here's some old code: https://github.com/GoogleCloudPlatform/app-maven-plugin/blob/7dafc0ee6d5798242aa08ee8b1fe9da8c9b45a8f/src/main/java/com/google/cloud/tools/maven/cloudsdk/CloudSdkMojo.java#L81

But maven compiler plugin compatibility for newer version of maven seem to be using a new config parameter `release` which is not accounted for in this code.